system-test/trace-express.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+if (!process.env.GCLOUD_PROJECT ||
+  !process.env.GOOGLE_APPLICATION_CREDENTIALS) {
+  throw new Error('system-test requires credentials to be available via ' +
+    'environment. Please set GCLOUD_PROJECT and ' +
+    'GOOGLE_APPLICATION_CREDENTIALS');
+}
+
+// trace-agent must be loaded before everything else.
+require('../').start({ flushDelaySeconds: 1 });
+
+const assert = require('assert');
+const googleAuth = require('google-auto-auth');
+const got = require('got');
+const queryString = require('querystring');
+const uuid = require('uuid');
+
+const WRITE_CONSISTENCY_DELAY_MS = 20 * 1000;
+const projectId = process.env.GCLOUD_PROJECT;
+
+function makeAuthorizedGot(auth) {
```

Comment:
I am not sure I follow how would that work? The authorized got function needs to return a promise. Are you suggesting getting rid of the got factory? That would incur multiple nesting levels in the code where I need to use `got`.

The code is dense, but it is readable if you understand promises.

Eventually this code can go into a separate module if we need to use this in other places.